### PR TITLE
Write documentation on creating custom CATcher sessions

### PR DESCRIPTION
**Pull request by luminousleek**
### Summary:
Fixes CATcher-org/CATcher#947

### Changes Made:
* Write a section in `dg/testing.md` detailing how to create custom sessions for CATcher
* Add a section on how to populate issue repositories with comments that conform to CATcher templates
* Edit `dg/user-workflow.md` to enable reuse

### Proposed Commit Message:
```
Write documentation on creating custom CATcher sessions

The instructions on loading custom sessions in the Testing page of the
DG does not specify how to create these sessions. This causes new
developers to get confused about the purpose of this section.

Let's add some documentation to explain how to create these custom
sessions, and also explain how to populate the repositories with issues.
```

### CODE REVIEW

**Review by seanlowjk — APPROVED** [review]
Seems clear and fine to me! @kkangs0226 any comments?

File: dg/testing.md [hunk 1]
```diff
@@ -12,13 +12,89 @@ This page contains information useful for testing of CATcher.
...
+  "phaseBugReporting": "bugreporting",
+  "phaseTeamResponse": "pe-results",
+  "phaseTesterResponse": "bugreporting",
+  "phaseModeration": "pe-evaluation"
+}
+```
+
+The `openPhases` property specifies the phases that are open. It is recommended to leave all the phases open for testing purposes. Do note that the Tutor Moderation phase is still under development. 
+
+The other four properties specify the name of the repositories that CATcher will store and retrieve from. In the example above, CATcher will retrive issues from the `bugreporting` repository for the Bug Reporting and Tester Response phases, and will retrive issues from the `pe-results` and `pe-evaluation` repositories for the Team Response and Tutor Moderation phases accordingly. If the repositories do not exist, CATcher will create them automatically.
```
Comment (kkangs0226) [inline dg/testing.md:39]: Typo for 'retrive' -> 'retrieve'

File: dg/testing.md [hunk 2]
```diff
@@ -12,13 +12,89 @@ This page contains information useful for testing of CATcher.
...
+
+**Issue Labels required:**
+- `severity.*`, e.g. `severity.Low`
+- `type.*`, e.g. `type.DocumentationBug`
+
+</panel>
+
+### Loading the custom session
+
+The custom session will not present in the default dropdown list on CATcher's startup page, so it can be loaded by clicking on the **file icon** beside the session dropdown. Following which, submit a file with the `.json` file extension, where the format is specified below.
```
Comment (kkangs0226) [inline dg/testing.md:93]: 'custom session will not **be** present'

**Review by kkangs0226 — COMMENTED** [review]
Thank you for working on this, it sounds much clearer now 👍 Have added 2 comments for minor typos

File: dg/testing.md [hunk 3]
```diff
@@ -12,13 +12,89 @@ This page contains information useful for testing of CATcher.
...
+  "phaseBugReporting": "bugreporting",
+  "phaseTeamResponse": "pe-results",
+  "phaseTesterResponse": "bugreporting",
+  "phaseModeration": "pe-evaluation"
+}
+```
+
+The `openPhases` property specifies the phases that are open. It is recommended to leave all the phases open for testing purposes. Do note that the Tutor Moderation phase is still under development. 
+
+The other four properties specify the name of the repositories that CATcher will store and retrieve from. In the example above, CATcher will retrieve issues from the `bugreporting` repository for the Bug Reporting and Tester Response phases, and will retrive issues from the `pe-results` and `pe-evaluation` repositories for the Team Response and Tutor Moderation phases accordingly. If the repositories do not exist, CATcher will create them automatically.
```
[the remark this answers is not in the document]
Comment (chunweii) [reply dg/testing.md:39]: >  and will retrive issues from the `pe-results` and `pe-evaluation` repositories

**Pull request merged**